set-up-location-driven-geotriggers/src/main/java/com/esri/arcgismaps/sample/setuplocationdrivengeotriggers/FeatureListAdapter.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.setuplocationdrivengeotriggers
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.arcgismaps.data.ArcGISFeature
+
+/**
+ * Provides a reference view holder to a custom view type
+ * for the FeatureListAdapter
+ */
+class FeatureViewHolder(view: View) : ViewHolder(view) {
+    // shows the title information
+    val titleView: TextView by lazy {
+        view.findViewById(R.id.featureNameTextView)
+    }
+}
+
+/**
+ * Custom RecyclerView Adapter to display the list of points of interests [featuresList]
+ * [onItemClickListener] event is called when an itemView clicks
+ */
+class FeatureListAdapter(
+    private val featuresList: MutableList<ArcGISFeature>,
```

Comment:
Interesting, I was not expecting the `List` and `MutableList` would reference the same collection.

Reply:
Yes it's a bit weird but it's a way of exposing a read-only reference to a mutable type. It's much more useful when working with SharedFlow and StateFlow (which we do all over the API) because you can have MutableSharedFlow/MutableStateFlow references internally and expose them as SharedFlow/StateFlow so users can collect but not push data of their own.

I think that's the intention of it but as a side effect it's possible to pass read-only references anywhere where you don't need a mutable one.